entities/backup/descriptor_test.go

```diff
@@ -479,77 +479,77 @@ func TestWildcardFunctionality(t *testing.T) {
 	// Test case scenarios for wildcard functionality
 	tests := []struct {
 		name            string
-		files           []string // Simulated list of file names
-		wildcardPattern string   // Wildcard pattern to match files
-		expectedMatch   []string // Expected list of files matching the pattern
-		expectedNoMatch []string // Expected list of files not matching the pattern
+		classes         []string // Simulated list of class names
+		wildcardPattern string   // Wildcard pattern to match class names
+		expectedMatch   []string // Expected list of class names matching the pattern
+		expectedNoMatch []string // Expected list of class names not matching the pattern
 	}{
 		{
 			name:            "Exact match",
-			files:           []string{"file1.txt", "file2.txt", "file3.txt"},
-			wildcardPattern: "file2.txt",
-			expectedMatch:   []string{"file2.txt"},
-			expectedNoMatch: []string{"file1.txt", "file3.txt"},
+			classes:         []string{"ClassA", "ClassB", "ClassC"},
+			wildcardPattern: "ClassB",
+			expectedMatch:   []string{"ClassB"},
+			expectedNoMatch: []string{"ClassA", "ClassC"},
 		},
 		{
 			name:            "Wildcard match",
-			files:           []string{"file1.txt", "file2.txt", "file3.txt"},
-			wildcardPattern: "file*.txt",
-			expectedMatch:   []string{"file1.txt", "file2.txt", "file3.txt"},
+			classes:         []string{"ClassA", "ClassB", "ClassC"},
+			wildcardPattern: "Class*",
+			expectedMatch:   []string{"ClassA", "ClassB", "ClassC"},
 			expectedNoMatch: []string{},
 		},
 		{
 			name:            "Pattern match with character wildcard",
-			files:           []string{"file1.txt", "file2.txt", "file3.txt"},
-			wildcardPattern: "file?.txt",
-			expectedMatch:   []string{"file1.txt", "file2.txt", "file3.txt"},
+			classes:         []string{"ClassA", "ClassB", "ClassC"},
+			wildcardPattern: "Class?",
+			expectedMatch:   []string{"ClassA", "ClassB", "ClassC"},
 			expectedNoMatch: []string{},
 		},
 		{
 			name:            "Pattern match with range wildcard",
-			files:           []string{"file1.txt", "file2.txt", "file3.txt", "file4.doc"},
-			wildcardPattern: "file[1-3].txt",
-			expectedMatch:   []string{"file1.txt", "file2.txt", "file3.txt"},
-			expectedNoMatch: []string{"file4.doc"},
+			classes:         []string{"Class1", "Class2", "Class3", "OtherClass"},
+			wildcardPattern: "Class[1-3]",
```

Comment:
I think this test is a bit misleading, it appears to satisfy a regex, (which wouldn't be a valid classname anyway). For example, based on the test case, one would probably assume that `"Class4"` wouldn't match, but given the implementation of the wildcard, it would.